### PR TITLE
Misc doc upload task updates

### PR DESCRIPTION
**Pull request by bryceAebi**
1. Enable user to select multiple files for doc upload
2. Fix css issue where collapsed task rows were covering the task rows below
3. Display uploaded document links in completed task
4. Enable user to delete the uploads to a task, moving the task back to the TODO state
5. Enable user to add/edit description on a document upload task

### CODE REVIEW

**Review by tomaszantas — COMMENTED** [review]
I had an issue to upload documents within Bank Transactions. Maybe I'm connected to old staging API, because Tasks response contains `document` field, not `documents`.

You can check [PR 437](https://github.com/Layer-Fi/layer-react/pull/437) - I left there few suggestions and comments. But please be aware that merging my PR into yours may not be safe.

Btw, what IDE are you using? If it's VSCode/Cursor, you can install eslint extension and add this settings.json file to the root folder:

`.vscode/settings.json`

```json
{
  "editor.formatOnSave": false,
  "editor.codeActionsOnSave": {
    "source.fixAll.eslint": "explicit"
  },
  "eslint.format.enable": true,
  "eslint.workingDirectories": [
    {
      "mode": "auto"
    }
  ],
  "prettier.enable": false,
  "editor.defaultFormatter": "dbaeumer.vscode-eslint",
  "emeraldwalk.runonsave": {
    "commands": [
      {
        "match": ".*",
        "isAsync": true,
        "cmd": "cd components && npm run build"
      }
    ]
  },
  "[jsonc]": {
    "editor.defaultFormatter": "vscode.json-language-features"
  }
}
```

It will auto-format files with eslint and rebuild components.

File: src/components/Tasks/Tasks.tsx [hunk 1]
```diff
@@ -24,7 +24,8 @@ export const UseTasksContext = createContext<UseTasksContextType>({
   error: undefined,
   refetch: () => {},
   submitResponseToTask: () => {},
-  uploadDocumentForTask: () => {},
+  uploadDocumentsForTask: () => {},
+  deleteUploadsForTask: () => {},
```
[the remark this answers is not in the document]
Comment (imkesin) [reply src/components/Tasks/Tasks.tsx:28]: Is this for compatibility? I didn't think the `Layer` object was a top-level export.

File: src/types/tasks.ts [hunk 2]
```diff
@@ -1,5 +1,17 @@
 import { DocumentType } from './file_upload'
 
+interface PresignedUrl {
+  presignedUrl: string
+  // add the other fields if/when necessary
+}
```
Comment (imkesin) [inline src/types/tasks.ts:6]: Can use the existing `S3PresignedUrl` type (interface?) instead

File: src/components/TasksListItem/TasksListItem.tsx [hunk 3]
```diff
@@ -75,30 +124,32 @@ export const TasksListItem = ({
...
               />
+            {task.user_response_type === 'UPLOAD_DOCUMENT' ? (
+              <div className="Layer__tasks-list__link-list">
+              {selectedFiles ? (
+                <div className="Layer__tasks-list__link-list-header">Selected Files:</div>
+                ) : task.documents ? (
+                <div className="Layer__tasks-list__link-list-header">Uploaded Files:</div>
+                ) : null}
+              <ul className="Layer__tasks-list__links-list">
+                {task.documents?.map((document, idx) => (
```
[the remark this answers is not in the document]
Comment (imkesin) [reply src/components/TasksListItem/TasksListItem.tsx:142]: I agree that I'm only seeing a singular `document` as a possible field on a task.

[the remark this answers is not in the document]
Comment (bryceAebi) [reply src/components/TasksListItem/TasksListItem.tsx:142]: Yes. I have a backend PR that is not yet merged

**Comment by bryceAebi** [comment]
@imkesin @tomaszantas thanks for the review. I have made the requested updates. You can view a full demo of the features in a screen recording I posted in #proj-doc-upload-tasks

File: src/components/Input/FileInput.tsx [hunk 4]
```diff
@@ -43,6 +45,7 @@ export const FileInput = ({
         </TextButton>
         <input
           type='file'
+          {...(allowMultipleUploads && { multiple: true})}
```
Comment (tomaszantas) [inline src/components/Input/FileInput.tsx:48]: Would this work?

```suggestion
          multiple={allowMultipleUploads}
```

File: src/components/Input/FileInput.tsx [hunk 5]
```diff
@@ -5,11 +5,12 @@ import { ButtonVariant } from '../Button/Button'
 
 export interface FileInputProps {
   text?: string
-  onUpload?: (file: File) => void
+  onUpload?: (files: File[]) => void
```
Comment (tomaszantas) [inline src/components/Input/FileInput.tsx:8]: `<FileInput />` component is used in BankTransactions-related components, like `BusinessForm`, where functions expect parameter to be a single file (`File`), and not a list of files (`File[]`). I would opt to update functions there with sth like this:

```tsx
onUpload={files => receiptsRef.current?.uploadReceipt(files[0])}
```

**Review by tomaszantas — COMMENTED** [review]
I can't find video above. In overall code looks good, but one thing is broken. FileInput is used in BankTransactions' receipts, where functions expect to receive one file, not a list of files. I left suggestion how functions could be changed to make them working.

Btw, you can run:

```bash
npm-dts generate -L debug
```

...in terminal to see if all types are set properly (eventually, we will have this check on CI too).

It will also produce `index.d.ts` file, which can be removed before pushing PR.

**Review by tomaszantas — APPROVED** [review]
LGTM!

**Pull request merged**